src/smalloc.h

```diff
@@ -85,13 +86,45 @@ NODE_EXTERN void Alloc(Environment* env,
  * Free memory associated with an externally allocated object. If no external
  * memory is allocated to the object then nothing will happen.
  */
-NODE_EXTERN void AllocDispose(Environment* env, v8::Handle<v8::Object> obj);
+NODE_EXTERN void AllocDispose(v8::Isolate* isolate, v8::Handle<v8::Object> obj);
 
 
 /**
  * Check if the Object has externally allocated memory.
  */
-NODE_EXTERN bool HasExternalData(Environment* env, v8::Local<v8::Object> obj);
+NODE_EXTERN bool HasExternalData(v8::Isolate* isolate,
+                                 v8::Local<v8::Object> obj);
+
+
+// Internal use
+void Alloc(Environment* env,
```

Comment:
If so then I'd recommend doing that in a different PR, and doing with all other headers that do the same (e.g. `node_buffer.h`).